src/vdbfusion/pybind/vdbfusion_pybind.cpp

```diff
@@ -70,41 +82,15 @@ PYBIND11_MODULE(vdbfusion_pybind, m) {
              "space_carving"_a = false)
         // TODO: add support for this
         .def("_integrate",
-             py::overload_cast<const std::vector<Eigen::Vector3d>&, const Eigen::Matrix4d&,
+             py::overload_cast<const std::vector<Eigen::Vector3d>&,
+                               const std::vector<Eigen::Vector3d>&, const Eigen::Matrix4d&,
                                const std::function<float(float)>&>(&VDBVolume::Integrate),
-             "points"_a, "extrinsic"_a, "weighting_function"_a)
+             "points"_a, "colors"_a, "extrinsic"_a, "weighting_function"_a)
         .def("_integrate",
-             py::overload_cast<const std::vector<Eigen::Vector3d>&, const Eigen::Vector3d&,
+             py::overload_cast<const std::vector<Eigen::Vector3d>&,
+                               const std::vector<Eigen::Vector3d>&, const Eigen::Vector3d&,
                                const std::function<float(float)>&>(&VDBVolume::Integrate),
-             "points"_a, "origin"_a, "weighting_function"_a)
-        .def(
```

Comment:
This is the lambda madness I was mentioning before

Reply:
I put some of it back. Looks like it is about 20%ish faster. I also managed to get it working using a `py::object` as the weighting_function argument and then do the conversion in C++/pybind11, but it isn't really much more elegant. I removed the option of having a 4x4 extrinsic matrix here and did the reduction to a "origin vector" in python. That cut the number of necessary overloads by a factor of 2.